packages/@aws-cdk/aws-glue/lib/mongodb-connection.ts

```diff
@@ -0,0 +1,83 @@
+import * as constructs from 'constructs';
+import { Connection, ConnectionOptions, ConnectionType } from './connection';
+
+/**
+ * Construction properties for {@link MongoDBConnection}
+ *
+ * @see https://docs.aws.amazon.com/glue/latest/dg/connection-mongodb.html
```

Comment:
Renamed to source connection. The differences am noticing are around `partitionerOptions` which I avoided adding coz it itself is a bunch of extra properties and documentation was not clear about how to best name them.

Reply:
@iliapolo fixed merge conflicts and now this is ready for another round of review... can't wait till we get to the `Job` one, it will be... mmm... something :-)

Reply:
What about `uri` that is mentioned as a required option in the below documentation link? And `username` and `password`?

Reply:
You are right, I missed those. However, I'm still not really sure about the right thing here
- https://docs.aws.amazon.com/glue/latest/dg/connection-mongodb.html: The connection `url`, `username` and `password` are stored in the MongoDB connection. Other options can be specified in the job script as additional options.
- https://docs.aws.amazon.com/glue/latest/dg/connection-mongodb.html: `uri` instead of `url`

What I did try is, using the console to create a mongodb connection. It prompted me for connection url which itself includes a databasename, username, password, and vpc configs.

I think I need to add an integ test to try a couple of variations to figure out the right properties to use to match what gets created in console... I can't say it enough, documentation is really horrible!

Reply:
Neither `url` or `uri` seem to work. I've also checked https://docs.aws.amazon.com/glue/latest/dg/aws-glue-api-catalog-connections.html#aws-glue-api-catalog-connections-ConnectionInput which lists a different set of potential values for properties keys.

What do you think about dropping the whole MongoDB connection in favour of the now ready-to-extend `Connection`? And maybe reserve those type-safe classes for when the documentation is clearer?

Reply:
After further discussions, we'll drop the MongoDB connection construct while keeping the escape hatches to ensure the idea can be still implemented once we have clearer requirements/documentation on each connection type requirements.